src/setup/parReader.cpp

```diff
@@ -1158,6 +1162,44 @@ void parseLinearSolver(const int rank, setupAide &options, inipp::Ini *par, std:
       p_solver = "PCG";
     }
   }
+  else if (p_solver.find("chebyshev") != std::string::npos) { // TODO: new
+    std::vector<std::string> list;
+    list = serializeString(p_solver, '+');
+    std::string n = "10";
+    std::string m = "20";
+    for (std::string s : list) {
+      const auto nvectorStr1 = parseValueForKey(s, "start");
+      if (!nvectorStr1.empty()) {
+        n = nvectorStr1;
+      }
+      const auto nvectorStr2 = parseValueForKey(s, "iter");
+      if (!nvectorStr2.empty()) {
+        m = nvectorStr2;
+      }
+    }
+    options.setArgs(parSectionName + "CHEBYSHEV START", n);
+    options.setArgs(parSectionName + "CHEBYSHEV ITER", m);
+
+
+    if (p_solver.find("block") != std::string::npos) {
+      options.setArgs(parSectionName + "BLOCK SOLVER", "TRUE");
+    }
+    else {
+      options.setArgs(parSectionName + "BLOCK SOLVER", "FALSE");
+    }
+    if (p_solver.find("extra") != std::string::npos) {
+      options.setArgs(parSectionName + "CHEBYSHEV EXTRA", "TRUE");
+    }
+    else {
+      options.setArgs(parSectionName + "CHEBYSHEV EXTRA", "FALSE");
+    }
+    if (p_solver.find("fcg") != std::string::npos || p_solver.find("flexible") != std::string::npos) {
+      p_solver = "CHEBYSHEV+FLEXIBLE"; // TODO: need test
```

Comment:
`fcg`? Search for `fcheb` or something like that instead.

I'd go ahead and mark this as _not_ supported -- we also don't have a use-case at the moment, right?

Reply:
Right, and I need more thinking whether it can be applied to flex-CG.